Sources/Stevenson/CircleCIService.swift

```diff
@@ -51,4 +55,90 @@ public struct CircleCIService {
             }
         }
     }
+
+}
+
+extension CircleCIService {
+
+    public struct PipelineRequest: Encodable {
+        let branch: String
+        let parameters: [String: Parameter]
+
+        public enum Parameter: Encodable {
+            case bool(Bool)
+            case string(String)
+
+            public func encode(to encoder: Encoder) throws {
+                var values = encoder.singleValueContainer()
+                switch self {
+                case let .bool(value):
+                    try values.encode(value)
+                case let .string(value):
+                    try values.encode(value)
+                }
+            }
+        }
+    }
+
+    public struct PipelineResponse: Content {
+        public let branch: String
+        public let buildURL: URL
+    }
+
+    struct PipelineID: Content {
+        let id: String
+    }
+
+    struct Pipeline: Content {
+        let workflows: [Workflow]
+        let vcs: VCS
+
+        struct Workflow: Content {
+            let id: String
+        }
+
+        struct VCS: Content {
+            let branch: String
+        }
+    }
+
+    private func pipelineURL(pipelineID: String) -> URL {
+        return URL(
+            string: "/api/v2/pipeline/\(pipelineID)?circle-token=\(token)",
+            relativeTo: baseURL
+        )!
+    }
+
+    private func workflowURL(workflowID: String) -> URL {
+        return URL(
+            string: "/workflow-run/\(workflowID)",
+            relativeTo: baseURL
+        )!
+    }
```

Comment:
I think it would make more sense and be more readable to implement that as a function directly on `Workflow` instead.

Something like:

```swift
struct Workflow: Content {
    let id: String
    func url(baseURL: URL) {
        return URL(string: "/workflow-run/\(self.id)", relativeTo: baseURL)!
    }
}
```

Reply:
this is the URL in the CircleCI dashboard, not directly related to the workflow data, so I'd keep it in the service in this case, and it's private anyway 🤷‍♂  It won't change anything significantly, will be just `$0.url(baseURL: self.baseURL)` instead of `self.workflowURL(workflowID: $0.id)` (if it would be our code base it would be even `^\.id >>> self.workflowURL`), I find later more readable actually